api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -127,19 +127,26 @@ public void setUserService(UserService userService) {
   }
 
   @Override
-  public ResponseEntity<ClusterListResponse> listClusters() {
+  public ResponseEntity<ClusterListResponse> listClusters(String projectName) {
     User user = this.userProvider.get();
+    // TODO as part of Billing Project epic RW-1205: enable status checking for arbitrary project
```

Comment:
My interpretation of RW-2476 is that implementing arbitrary billing project creation, configuration, and monitoring is not in scope, implying that no functional change is desired here.

Please let me know if you disagree!

Reply:
Agreed the things listed here are out of scope, but to slightly future proof this and avoid having to revisit this in the next sprint, I would suggest instead:

```
if (user.getFreeTierBillingProjectName() != null &&
    user.getFreeTierBillingProjectName().equals(projectName) &&
    user.getFreeTierBillingProjectStatusEnum() != BillingProjectStatus.READY) {
  throw new FailedPreconditionException(
          "User billing project is not yet initialized, cannot list/create clusters");
}
```